docs:  adding remote sse deeplinks (#2535)
block/goose: documentation/docs/getting-started/using-extensions.md

  </TabItem>
</Tabs>

### Deeplinks

> Deeplinks provide a convenient way to install and configure remote extensions in Goose Desktop. They use a `goose://` URL scheme with an `id`, `name` and `description` argument, and either a `url` argument or `cmd` and `arg` arguments.

Reviewer: feel like we can keep this part succinct because we will explain the different components of the URL after this, so it's a little redundant imo. 

```suggestion
Extensions can be installed using Goose's deep link protocol. The URL format varies based on the extension type:
```

<Tabs groupId="interface">
  <TabItem value="stdio" label="StandardIO" default>
```
goose://extension?cmd=<command>&arg=<argument>&id=...&name=...&description=...
```

> Parameters:
> - `cmd`: The base command to run (e.g., `npx`)
> - `arg`: (cmd only) Command arguments (can be repeated for multiple arguments: `&arg=...&arg=...&arg=...`)
> - `id`: Unique identifier for the extension
> - `name`: Display name for the extension
> - `description`: Brief description of the extension's functionality
> 

Reviewer: We could make it more clear which ones are required vs optional 
```suggestion
Required parameters:
 - `cmd`: The command to run (e.g., `npx`, `uvx`)
 - `name`: Display name for the extension
 - `description`: Brief description of the extension

 Optional parameters:
  - `arg`: Command arguments (can be specified multiple times)
  - `timeout`: Maximum time in seconds to wait for extension responses
```

A command like `npx -y @modelcontextprotocol/server-github` would be represented as:

```
goose://extension?cmd=npx&arg=-y&arg=%40modelcontextprotocol/server-github&id=...
```

Note that each parameter to the `npx` command is passed as a separate `arg` parameter in the deeplink.
  </TabItem>

  <TabItem value="sse" label="Server-Sent Events">
```

> goose://extension?url=<remote-sse-url>&id=...&name=...&description=...

Reviewer: ```suggestion
 goose://extension?cmd={command}&name={extension_name}&description={description}
```

```

> Parameters:
> - `url`: The URL of the remote SSE server
> - `id`: Unique identifier for the extension
> - `name`: Display name for the extension
> - `description`: Brief description of the extension's functionality

Reviewer: missing the timeout params as optional

```suggestion
 Required parameters:
 - `url`: The SSE endpoint URL
 - `name`: Display name for the extension
 - `description`: Brief description of the extension

Optional parameters:
 - `timeout`: Maximum time in seconds to wait for extension responses
```

For example, a deeplink for a URL like `http://localhost:8080/sse` would look like this when URL-encoded:

```

> goose://extension?url=http%3A%2F%2Flocalhost%3A8080%2Fsse&id=...&name=...&description=...

Reviewer: again i think the ellipsis could be misleading or hard to read
```suggestion
goose://extension?url={sse_endpoint}&name={extension_name}&description={description}
```

Note that you'll need [Node.js](https://nodejs.org/) installed on your system to run this command, as it uses `npx`.
:::

### Remote Extensions over SSE

<Tabs groupId="interface">
  <TabItem value="ui" label="Desktop" default>
To enable a remote extension over SSE, you can use a deeplink as [described above](#deeplinks).

Clicking on a `goose://` deeplink will open Goose Desktop and prompt you to install the extension. You can use the Advanced Settings menu to view and manage the installed extension.

For example, to use remote extension running on localhost on port 8080 like `http://localhost:8080/sse`, your deeplink would look like this:

```
goose://extension?url=http%3A%2F%2Flocalhost%3A8080%2Fsse&id=...&name=...&description=...
```

</TabItem>

> <TabItem value="cli" label="CLI">

Reviewer: revert this to it's prior state, please and thank you!
